File: 2583.cpp
#include <bits/stdc++.h>
using namespace std;
#define X first
#define Y second
#define endl '\n'
int board[100][100];
bool dist[100][100];
int dx[4] = {-1,0,1,0};
int dy[4] = {0,1,0,-1};
int m,n,k;
int box = 0;
vector<int> v;
queue<pair<int,int>> Q;
int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    memset(dist,false,sizeof(dist));
    memset(board,0,sizeof(board));
    int x1,y1,x2,y2;
    cin >> m >> n >> k;
    for(int i = 0; i < k; i++) {
        cin >> x1 >> y1 >> x2 >> y2;

        for(int j = x1; j < x2; j++) {
            for(int p = y1; p < y2; p++) {
                board[p][j] = 1;
            }
        }
    }
    
    for(int i = 0; i < m; i++) {
        for(int j = 0; j < n; j++) {
            cout << board[i][j] << ' ';
        }
        cout << endl;
    }
    cout << endl;
    for(int i = 0; i < m; i++) {
        int cnt = 0;
        for(int j = 0; j < n; j++) {
            if(board[i][j] == 1) continue;
            
            Q.push({i,j});
            dist[i][j] = true;
            while(!Q.empty()) {
            auto cur = Q.front(); Q.pop();
            for(int dir = 0; dir < 4; dir++) {
                int nx = cur.X + dx[dir];
                int ny = cur.Y + dy[dir];
                if(nx < 0 || nx >= m || ny < 0 || ny >= n) continue;
                if(board[nx][ny] == 1 || dist[nx][ny] == true) continue;
                cnt++;
                dist[nx][ny] = true;
                Q.push({nx,ny});
                }
            }
            
        }
        v.push_back(cnt);
    } 
    cout << box << endl;
    sort(v.begin(), v.end());
    for(int i = 0; i < v.size(); i++) {
        cout << v[i] << ' ';
    }
}

File: Python/BOJ/2583.py
m,n,k = map(int,input().split())
board = [[0] * n for _ in range(m)]
dist = [[False] * n for _ in range(m)]
dx = [0,0,1,-1]
dy = [1,-1,0,0]
v = []
for _ in range(k):
    x1,y1,x2,y2 = map(int,input().split())
    for i in range(x1, x2) :
        for j in range(y1,y2):
            board[j][i] = 1

cnt = 0
for i in range(m):
    for j in range(n):
        if board[i][j] or dist[i][j] : continue
        cnt += 1
        box = 0
        queue = [[i,j]]
        dist[i][j] = 1
        while queue:
            x,y = queue[0][0],queue[0][1]
            del queue[0]
            for k in range(4):
                nx = x + dx[k]
                ny = y + dy[k]
                if nx < 0 or nx >= m or ny < 0 or ny >= n : continue
                if dist[nx][ny] or board[nx][ny] : continue
                dist[nx][ny] = True
                queue.append([nx,ny])
                box += 1
        v.append(box+1)

v = sorted(v)

print(cnt)
for i in v:
    print(i, end=' ')